specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json

```diff
@@ -526,7 +526,7 @@
             "required": true,
             "x-ms-client-flatten": true,
             "schema": {
-              "$ref": "#/definitions/NameAndUserDataContract"
+              "$ref": "#/definitions/MetaDataContract"
```

Comment:
I doubt whether this would be a breaking change for SDK users. Please double check.

Reply:
OK.

Reply:
Consider code that looked like:
```C#
var parameter = NameAndUserDataContract(...);
client.PersonGroup.Create(body=parameter);
```
You now have changed the type of the parameter to something that derives from NameAndUserContract. As this is an input type, the code above won't compile after the change.

Reply:
Hi @johanste , we have generated the [SDK ](https://github.com/acured/azure-sdk-for-net/pull/4/files)from this Swagger and please have a look. The contract `NameAndUserDataContract` will never be used as a parameter for operations so your example code is unrealistic for actual users. `NameAndUserDataContract` exists only inside operations.

Reply:
@lebronJ, it is great that you are looking at the generated client libraries. However, there are two things to keep in mind when doing so; 1 - the swagger is not only used to feed autorest and generate client libraries. It is also used to generate reference documentation. For input types like this, it is usually not a concern. 2 - C# is not the only language that we generate clients for. Your assessment on this not being breaking is relying on autorest specific extensions (flattening) and exactly how autorest will generate C# code for this specific method. For other languages, this will not be the case (Go comes to mind). This is more of my concern.

And to expand to this, this is a breaking change for your C# library as well. You have inserted a positional parameter in the method signature. Consider the (previously valid) code:

```C#
client.CreateAsync("somefacelistid", "somename", new CancellationTokenSource().Token)
```

Reply:
@johanste Thanks a lot for your nice explanation. I understand this will be a breaking change for SDKs in some aspects. Let's discuss this in the email loop for next steps.